Deprecate FastMalloc macro for new types

Achierius opened the pull request:
#### d9db853f14ab390f3e69c476b831ae89b8e7f120
<pre>
Deprecate FastMalloc macro for new types
<a href="https://bugs.webkit.org/show_bug.cgi?id=295749">https://bugs.webkit.org/show_bug.cgi?id=295749</a>
<a href="https://rdar.apple.com/155564529">rdar://155564529</a>

Reviewed by Devin Rousso.

As part of our ongoing effort to move over to TZones, we want to make it
obvious to developers that the WTF_MAKE_FAST_ALLOCATED macro-family is
not to be used except when necessary. By adding &quot;DEPRECATED&quot; to the
name, this patch should hopefully give people second thoughts on using
the macro.

Developers should prefer to use the WTF_MAKE_TZONE_ALLOCATED family of
macros whenever possible; broadly, we expect them to be equi-performant
while providing security benefits beyond FastMalloc.
However, since TZone does not handle variable-sized types, this may not
be possible everywhere. If it is not possible to use TZone for a given
type, or if migrating it over shows a measurable performance regression,
then FastMalloc can be used for the time being -- it is still preferable
to using the system allocator.

Beyond that, this also changes the _FAST_ macro family to take the
type-name as a parameter: this should make it easier to migrate the
remaining objects when we do get around to it.

Canonical link: <a href="https://commits.webkit.org/297491@main">https://commits.webkit.org/297491@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8c9cd02b8acad3f048537f92f39dad4ea36e4f59

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/111953 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/31623 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/22109 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/117975 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/62187 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/113915 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/32300 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/40201 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/85071 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/35734 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/114900 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/25792 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/100744 "Passed tests") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/65503 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/111385 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/25117 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/18878 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/61823 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/104441 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/95185 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/18954 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/121285 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/110512 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/38986 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/29005 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/93912 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/39367 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/96996 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/93729 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/23927 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/38918 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/16713 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/35008 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/38880 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/44392 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/134772 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/38517 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/36271 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/41845 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/40233 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

dcrousso commented on Source/update_macros.py:
```diff
@@ -0,0 +1,95 @@
+    "WTF_DEPRECATED_MAKE_FAST_ALLOCATED_WITH_HEAP_IDENTIFIER",
+    "WTF_DEPRECATED_MAKE_STRUCT_FAST_ALLOCATED_WITH_HEAP_IDENTIFIER",
+    "WTF_DEPRECATED_MAKE_FAST_COMPACT_ALLOCATED_WITH_HEAP_IDENTIFIER",
+    "WTF_DEPRECATED_MAKE_STRUCT_FAST_COMPACT_ALLOCATED_WITH_HEAP_IDENTIFIER"
```
since this script seems to be focused specifically on these macros, id suggest either a) adjusting this script to take an arbitrary macro name as an argument so that it can be used for other things or b) renaming it to be more focused to them (e.g. `update_fast_allocated_macros.py`) or c) removing it entirely

dcrousso replied:
also, please move it to `Tools/Scripts` as it probably should not be inside `Source`

Achierius replied:
Thanks for the catch -- I didn't intend to commit this, will remove.

dcrousso commented on Source/update_macros.py:
```diff
@@ -0,0 +1,95 @@
+    
+    return None
+
+def update_macro_usage(file_path, content):
```
the style checker bot has some issues with this file, so if you do intend to check it in then please fix them

dcrousso reviewed (approved):
neat!

Achierius committed:
Deprecate FastMalloc macro for new types
https://bugs.webkit.org/show_bug.cgi?id=295749
rdar://155564529

Reviewed by Devin Rousso.

As part of our ongoing effort to move over to TZones, we want to make it
obvious to developers that the WTF_MAKE_FAST_ALLOCATED macro-family is
not to be used except when necessary. By adding "DEPRECATED" to the
name, this patch should hopefully give people second thoughts on using
the macro.

Developers should prefer to use the WTF_MAKE_TZONE_ALLOCATED family of
macros whenever possible; broadly, we expect them to be equi-performant
while providing security benefits beyond FastMalloc.
However, since TZone does not handle variable-sized types, this may not
be possible everywhere. If it is not possible to use TZone for a given
type, or if migrating it over shows a measurable performance regression,
then FastMalloc can be used for the time being -- it is still preferable
to using the system allocator.

Beyond that, this also changes the _FAST_ macro family to take the
type-name as a parameter: this should make it easier to migrate the
remaining objects when we do get around to it.

Canonical link: https://commits.webkit.org/297491@main
